### DISCUSSION TITLE
ScriptReference/GUILayoutUtility.GetRect

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#124] [Notes]
# ScriptReference/GUILayoutUtility.GetRect

https://docs.unity3d.com/ScriptReference/GUILayoutUtility.GetRect.html

**Comment by docscomments** [comment]
**Comment by RobD6:**  

If used in an EditorGUI (and possible non-editor too), the "Fixed" sized version of this (where you specify min and max) seems to happily overshoot the max width, instead expanding to the width of the space it's in. Adding [`GUILayout.MaxWidth`](https://docs.unity3d.com/ScriptReference/GUILayout.MaxWidth.html) as an option fixes this.  
For example, if you want a rect of 100x100  
`GetRect(100, 100, 100, 100)` will often give you back a width more than 100.  
`GetRect(100, 100, GUILayout.MaxWidth(100))` will actually work.

**Comment by FreyaHolmer** [comment]
During the Layout event, this returns `new Rect(0,0,1,1)`, so if you want to use this to calculate a rectangle for `GUILayout.BeginArea()`, you'll need its value when *not* in the layout event, and cache it

```cs
Rect rectViewport = Rect.zero;

void OnGUI() {
	// GetRect has to be called in Layout still, not just repaint:
	Rect r = GUILayoutUtility.GetRect( GUIContent.none, EditorStyles.textArea );
	if( Event.current.type == EventType.Repaint )
		rectViewport = r; // cache

	GUILayout.BeginArea( rectViewport, EditorStyles.textArea );
	// wow inside an area
	GUILayout.EndArea();
}
```
